controllers/apps/operations/reconfigure.go

```diff
@@ -113,65 +114,186 @@ func (r *reconfigureAction) syncDependResources(reqCtx intctrlutil.RequestCtx, c
+		opsRequest.Status.Progress = fmt.Sprintf("%d/%d", succeedCount, expectedCount)
+	}
+	if err := PatchOpsStatusWithOpsDeepCopy(reqCtx.Ctx, cli, resource, opsDeepCopy, phase); err != nil {
 		return "", 30 * time.Second, err
 	}
-	configSpec := ops.Reconfigure.Configurations[0]
+	return phase, 30 * time.Second, nil
+}
+
+func (r *reconfigureAction) doSyncReconfigureStatus(params reconfigureParams) (appsv1alpha1.OpsPhase, error) {
+	if isNoChange(params.configurationStatus.Conditions) {
+		params.reqCtx.Log.
+			WithValues("clusterName", params.clusterName).
+			WithValues("componentName", params.configurationItem.Name).
+			Info("the reconfiguring operation has no change!")
+		return appsv1alpha1.OpsSucceedPhase, nil
+	}
+
+	configSpec := params.configurationItem
+	resource, err := r.syncDependResources(params.reqCtx,
+		params.cli, params.resource, configSpec, params.componentName)
+	if err != nil {
+		return "", err
+	}
+
 	item := resource.ConfigurationObj.Spec.GetConfigurationItem(configSpec.Name)
 	itemStatus := resource.ConfigurationObj.Status.GetItemStatus(configSpec.Name)
 	if item == nil || itemStatus == nil {
-		return appsv1alpha1.OpsRunningPhase, 30 * time.Second, nil
+		return appsv1alpha1.OpsRunningPhase, nil
 	}
 
 	switch phase := reconfiguringPhase(resource, *item, itemStatus); phase {
 	case appsv1alpha1.CCreatingPhase, appsv1alpha1.CInitPhase:
-		return appsv1alpha1.OpsFailedPhase, 0, core.MakeError("the configuration is creating or initializing, is not ready to reconfigure")
+		return appsv1alpha1.OpsFailedPhase, core.MakeError("the configuration is creating or initializing, is not ready to reconfigure")
 	case appsv1alpha1.CFailedAndPausePhase:
-		return syncStatus(cli, reqCtx, opsRes, itemStatus, phase, appsv1alpha1.OpsFailedPhase, appsv1alpha1.ReasonReconfigureFailed)
+		return appsv1alpha1.OpsFailedPhase,
+			syncStatus(params.configurationStatus, params.resource, itemStatus, phase)
 	case appsv1alpha1.CFinishedPhase:
-		return syncStatus(cli, reqCtx, opsRes, itemStatus, phase, appsv1alpha1.OpsSucceedPhase, appsv1alpha1.ReasonReconfigureSucceed)
+		return appsv1alpha1.OpsSucceedPhase,
+			syncStatus(params.configurationStatus, params.resource, itemStatus, phase)
 	default:
-		return syncStatus(cli, reqCtx, opsRes, itemStatus, phase, appsv1alpha1.OpsRunningPhase, appsv1alpha1.ReasonReconfigureRunning)
+		return appsv1alpha1.OpsRunningPhase,
+			syncStatus(params.configurationStatus, params.resource, itemStatus, phase)
 	}
 }
 
 func (r *reconfigureAction) Action(reqCtx intctrlutil.RequestCtx, cli client.Client, resource *OpsResource) error {
-	var (
-		opsRequest    = resource.OpsRequest
-		spec          = &opsRequest.Spec
-		clusterName   = spec.ClusterRef
-		componentName = spec.Reconfigure.ComponentName
-		reconfigure   = spec.Reconfigure
-	)
-
-	if !needReconfigure(opsRequest) {
+	opsRequest := resource.OpsRequest.Spec
+	if opsRequest.Reconfigure != nil && len(opsRequest.Reconfigure.Configurations) > 0 {
```

Comment:
Can we merge a single component into a element of multiple components from the beginning, so that we can use a set of code？

Reply:
done